src/metrics.ts

```diff
@@ -745,7 +745,7 @@ export class DependencyPins extends BaseMetric {
 
 		for (const dependency in dependencies) {
 			const version = dependencies[dependency];
-			if (/^(?:\d+\.\d+\.\d+|\d+\.\d+(\.[\d+*Xx])?|~\d+\.\d+(\.\d+)?|\^0\.\d+(\.\d+)?|\d+\.\d+(\.[*Xx])?)$/.test(version)) {
+			if (/^(?:\d+\.\d+\.\d+|\d+\.\d+(\.[\d+\*Xx])?|~\d+\.\d+(\.\d+)?|\^0\.\d+(\.\d+)?|\d+\.\d+(\.[\*Xx])?)$/.test(version)) {
```

Comment:
should be: `if (/^(?:\d+\.\d+\.\d+|\d+\.\d+(\.[\d+*Xx])?|~\d+\.\d+(\.\d+)?|\^0\.\d+(\.\d+)?|\d+\.\d+(\.[*Xx])?)$/.test(version)) {`